homeassistant/components/media_player/sonos.py

```diff
@@ -62,6 +62,11 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     if discovery_info:
         player = soco.SoCo(discovery_info)
+
+        # if device allready exists by config
+        if player.uid in [device.player_uid for device in DEVICES]:
```

Comment:
Entity implements [an equals method](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py#L281-L284) based on `unique_id`. So you can write:

```python
if player in DEVICES:
    return True
```

However, another option is that you leave the filtering up to the EntityComponent who does [exactly the same check](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_component.py#L172-L173)